src/nidcpower/custom_types/lcr_load_compensation_spot.py

```diff
@@ -73,6 +73,9 @@ def __init__(self, frequency, reference_value_type, reference_value):
         self.reference_value_type = enums.LCRReferenceValueType(reference_value_type)
         self.reference_value = reference_value
 
+    def create_copy(self, target_class):
+        return target_class(frequency=self.frequency, reference_value_type=self.reference_value_type.value)
```

Comment:
@DavidCurtiss this was the issue you and I ran into a couple of days ago with the `reference_value` property, which shows up as `reference_value_a` and `reference_value_b` in the proto. The offending test (`test_perform_lcr_load_compensation`) passes if we don't supply the reference_value at all, like I have here, but I don't think that's what we want because I can see that the test tries plugging in several reference_values. I'm looking into a better solution.

Reply:
Hmm, well... now we're correctly pulling out the real and imaginary components, but I don't know that this is exactly how we want the test cases with non-complex numbers to work. Or maybe it is? In any case, it seems that we can't access a property called `reference_value` from this scope.

Reply:
David and I discussed this more, and we decided to add error handling to `create_copy()`, so that the function handles both complex and non-complex inputs in a clearer way.